modules/firebase/firebase_test.go

```diff
@@ -0,0 +1,74 @@
+package firebase_test
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/firestore"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/firebase"
+)
+
+func TestFirebase(t *testing.T) {
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(3*time.Minute))
+	defer cancel()
+
+	ctr, err := firebase.Run(ctx, "ghcr.io/u-health/docker-firebase-emulator:13.29.2",
+		firebase.WithRoot(filepath.Join(".", "firebase")),
+	)
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+
+	// perform assertions
+	// Ports are linked to the example config in firebase/firebase.json
+
+	firestoreUrl, err := ctr.ConnectionString(ctx, "8080/tcp")
+	assert.NoError(t, err)
+	assert.NotEmpty(t, firestoreUrl)
+	_ = os.Setenv("FIRESTORE_EMULATOR_HOST", firestoreUrl)
+	c, err := firestore.NewClient(ctx, firestore.DetectProjectID)
+	require.NoError(t, err)
+	defer c.Close()
+	w, err := c.Collection("example").Doc("one").Set(ctx, map[string]interface{}{
+		"foo": "bar",
+	})
+	require.NotNil(t, w)
+	require.NoError(t, err)
+	snap, err := c.Collection("example").Doc("one").Get(ctx)
+	require.NoError(t, err)
+	var out map[string]string
+	err = snap.DataTo(&out)
+	require.NoError(t, err)
+	assert.Equal(t, "bar", out["foo"])
+}
+
+func TestFirebaseBadDirectory(t *testing.T) {
```

Comment:
question: Is this issue you were referencing as in you are expecting:
`./failure` to be copied to `/srv/firebase` and instead if ends up as `/srv/failure`?

Reply:
Yes, exactly.

And if I attempt to specify the destination path as `/srv/firebase/` I get either `/srv/firebase/firebase` or `/srv/firebase/failure`, neither of which work for my use case.

I could attempt to modify the container launch script to accommodate for this, but `firebase` executable expects the configuration to be in the launch dir. That means, I would have to stray away from default execution method, and other containers would be less likely to work with this adapter.

Reply:
Thanks, I was looking for a good example of this in real world.